### PR TITLE
[AMDGPU] Exclude implicit uses and defs in SIInsertWaitcnts. NFC.

### PR DESCRIPTION
**Pull request by rampitec**
Change all_uses/all_defs to uses/defs where it obviously cannot happen.

### CODE REVIEW

File: llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp [hunk 1]
```diff
@@ -863,15 +863,15 @@ void WaitcntBrackets::updateByEvent(const SIInstrInfo *TII,
...
+        for (MachineOperand &DefMO : Inst.defs()) {
           if (TRI->isVGPR(*MRI, DefMO.getReg())) {
             setRegScore(
                 TRI->getEncodingValue(AMDGPU::getMCReg(DefMO.getReg(), *ST)),
                 EXP_CNT, CurrScore);
           }
         }
       }
-      for (const MachineOperand &Op : Inst.all_uses()) {
+      for (const MachineOperand &Op : Inst.uses()) {
```
Comment (jayfoad) [inline llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp:874]: Here it's not obvious (to me) what kind of instruction `Inst` is.

File: llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp [hunk 2]
```diff
@@ -2326,7 +2326,7 @@ bool SIInsertWaitcnts::shouldFlushVmCnt(MachineLoop *ML,
         if (MI.mayStore())
           HasVMemStore = true;
       }
-      for (const MachineOperand &Op : MI.all_uses()) {
+      for (const MachineOperand &Op : MI.uses()) {
```
Comment (jayfoad) [inline llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp:2329]: Here also, I think `MI` could be anything.

**Review by jayfoad — COMMENTED** [review]
> Change all_uses/all_defs to uses/defs where it obviously cannot happen.

I don't understand. Do you mean it is obvious that these instructions have no implicit operands? Or you somehow know that they are not important ones?

File: llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp (hunk 1 above)
Comment (rampitec) [reply llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp:874]: We do not define implicit VGPR uses anywhere, except for super-reg liveness purposes. This is my assumption here. The next line checks it is a VGPR, so this is not some of the special registers. Am I wrong in that assumption?

File: llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp (hunk 2 above)
Comment (rampitec) [reply llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp:2329]: Same here, I am excluding VGPR implicit uses, which are never real. Besides the whole function is an opportunistic optimization.

**Comment by rampitec** [comment]
So the assumption here an implicit VGPR use is never a real use.

**Comment by rampitec** [comment]
> So the assumption here an implicit VGPR use is never a real use.

Actually the assumption is not correct. At least movrels/movreld uses implicit VGPRs which a real.

**Pull request closed without merging**